### PR TITLE
DOP-1899: Add Segment tracking on codeblock

### PR DESCRIPTION
**Pull request by rayangler**
[[Jira ticket](https://jira.mongodb.org/browse/DOP-1899)]
[[Realm staging](https://docs-mongodbcom-staging.corp.mongodb.com/master/realm/raymundrodriguez/DOP-1899/deploy/create-with-cli)]

### CODE REVIEW

File: src/components/Code.js [hunk 1]
```diff
@@ -84,9 +90,9 @@ const Code = ({
         `}
         highlightLines={emphasizeLines}
         language={getLanguage(lang)}
+        onCopy={reportCodeCopied}
         showLineNumbers={linenos}
         showWindowChrome={caption ? true : false}
-        chromeTitle={caption ? caption : ''}
```
Comment (sophstad) [inline src/components/Code.js:89]: Alphabetization 😊👍

File: src/components/Code.js [hunk 2]
```diff
@@ -54,6 +55,10 @@ const Code = ({
     if (isBrowser) code = htmlDecode(code);
   }
 
+  const reportCodeCopied = useCallback(() => {
+    reportAnalytics('CodeblockCopied', { code });
```
Comment (sophstad) [inline src/components/Code.js:59]: This event was [previously](https://github.com/mongodb/snooty/blob/0c2d67e316910218ed44469191e17a9410525880/src/components/Code.js#L124) named `Codeblock Copied` — nitpicky I know but just want to ensure this is the event name Elle wants! 😄

File: src/components/Code.js (hunk 1 above)
Comment (sophstad) [inline src/components/Code.js:89]: Alphabetization 😊👍

**Review by sophstad — APPROVED** [review]
Just one note, looks great to me!

File: src/components/Code.js (hunk 2 above)
Comment (rayangler) [reply src/components/Code.js:59]: The Jira ticket title stated it should be `CodeblockCopied` so I just went with that

Comment (rayangler) [reply src/components/Code.js:59]: @eshwer Do we want the event for this to be `Codeblock Copied` like the previous version, or `CodeblockCopied`?

Comment (eshwer) [reply src/components/Code.js:59]: let's do the latter please (no space)

**Pull request merged**